src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs

```diff
@@ -202,24 +202,26 @@ internal static void CheckForSevereException(Exception e)
 
             System.Threading.Thread.CurrentThread.Name = "ConsoleHost main thread";
 
-            s_theConsoleHost = ConsoleHost.CreateSingletonInstance(configuration);
-            s_theConsoleHost.BindBreakHandler();
-
-            PSHost.IsStdOutputRedirected = Console.IsOutputRedirected;
-
-            if (args == null)
+            // We might be able to ignore console host creation error if we are running in 
+            // server mode, which does not require a console.
+            HostException hostException = null;
+            try
             {
-                args = new string[0];
+                s_theConsoleHost = ConsoleHost.CreateSingletonInstance(configuration);
```

Comment:
Shouldn't this be within the try-finally on line 222 since the finally conditionally disposes it?